### PR TITLE
Basic validation for select widget

### PR DESCRIPTION
**Pull request by KaneFreeman**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Add basic validation to the select widget, along with an `onValidate` event to match `TextInput`.

**Use case**: checking the validity of a whole form. Currently validation logic for select needs to be written each time the widget is used (or a wrapper widget needs to be written).

### CODE REVIEW

File: src/select/index.ts [hunk 1]
```diff
@@ -149,6 +157,40 @@ export class Select extends ThemedMixin(FocusMixin(WidgetBase))<SelectProperties
...
+		const {
+			_state: state,
+			properties: { onValidate, value, required }
+		} = this;
+
+		if (!onValidate) {
+			return;
+		}
+
+		if (
```
Comment (tomdye) [inline src/select/index.ts]: if this widget has no value, but it is required then should it not report as `invalid` once it's been interacted with?

Comment (KaneFreeman) [reply src/select/index.ts]: This has been addressed. I have also matched the logic to be closer to TextInput and TextArea.

**Comment by agubler** [comment]
@tomdye is this outdated now that we are working on the replacement for `Select`?

**Comment by tomdye** [comment]
I will shortly be re-writing Select to use the new menu / popup functionality so will bake this into the new code rather than landing this PR. Thanks

**Pull request closed without merging**